lib/AST/AccessRequests.cpp

```diff
@@ -81,7 +81,7 @@ AccessLevelRequest::evaluate(Evaluator &evaluator, ValueDecl *D) const {
   // Special case for dtors and enum elements: inherit from container
   if (D->getKind() == DeclKind::Destructor ||
       D->getKind() == DeclKind::EnumElement) {
-    if (D->hasInterfaceType() && D->isInvalid()) {
```

Comment:
I kinda wonder why this path has to exist at all.  I bet if you drop it you could preserve the “syntactic algorithm” that @johnfairh is asking for.

Reply:
I think what this means to ask is “is this enum element or destructor actually nested in a nominal type”.